numpy/core/arrayprint.py

```diff
@@ -59,11 +61,16 @@
     'formatter': None,
     # Internally stored as an int to simplify comparisons; converted from/to
     # str/False on the way in/out.
-    'legacy': sys.maxsize}
+    'legacy': sys.maxsize,
+    'exp_format': False,  # force exp formatting as Python do when ".2e"
+    'trim': '.'  # Controls post-processing trimming of trailing digits
+                 # see Dragon4 arguments at `format_float_scientific` below
```

Comment:
The exp_format and trim kwargs should go before legacy, since that is the way the implementation has them